packages/dds/tree/src/feature-libraries/persisted-config/persistedConfig.ts

```diff
@@ -0,0 +1,336 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/common-utils";
+import { ISequencedDocumentMessage } from "@fluidframework/protocol-definitions";
+
+// TODO: Document, decide if semver-like even provides value here (over e.g. plain-old number).
+export type Version = `${number}.${number}.${number}${"-internal" | ""}`;
+
+export interface PersistedConfig {
+	version: Version;
+	// Strictly increasing number which identifies the iteration of the proposed protocol. It's suggested
+	// that applications are able to modify this using the same mechanisms with which they can modify a DDS's
+	// configuration, which enables safe rollback policies.
+	// Example: application enables feature flag 1 in iteration 2, then realizes there is a problem with it.
+	// application then disables feature flag 1 in iteration 3.
+	// Using a single version number and monotonic increase on it isn't sufficient: DDSes may support configuration
```

Comment:
But, if all you had was the version, wouldn't you just keep bumping the version every time you make a protocol change? I don't understand what changes by adding `protocolIteration`. Unless you're saying that the version specifies "what capabilities does this config have" and protocolIteration says "which of these capabilities are we using right now"?

Reply:
The motivation for separating the iteration from the agreed upon config is it enables safe velocity principles--say you're rolling out a compression feature in v2, but find out there's a problem with it. Provided the problem doesn't corrupt data in certain ways, you have a chance of flipping a killswitch by releasing a new config with *higher* protocol iteration but v1 format.

Reply:
Why do we want a higher iteration protocol?

Reply:
basically a client making an edit to a session needs to know if their version of the persisted config is newer than the one that's currently agreed upon. you can't know this based only on the format when accounting for potential desire to revert to an older format.

Reply:
protocol is probably not a great term anyway. Perhaps `formatVersion` and `configVersion` with clearly documented instructions that configVersion should always increment most clearly communicates the intent.